plugins/kafka/package.json

```diff
@@ -0,0 +1,53 @@
+{
+  "name": "@backstage/plugin-kafka",
+  "version": "0.1.0",
+  "main": "src/index.ts",
+  "types": "src/index.ts",
+  "license": "Apache-2.0",
+  "publishConfig": {
+    "access": "public",
+    "main": "dist/index.esm.js",
+    "types": "dist/index.d.ts"
+  },
+  "scripts": {
+    "build": "backstage-cli plugin:build",
+    "start": "backstage-cli plugin:serve",
+    "lint": "backstage-cli lint",
+    "test": "backstage-cli test",
+    "diff": "backstage-cli plugin:diff",
+    "prepack": "backstage-cli prepack",
+    "postpack": "backstage-cli postpack",
+    "clean": "backstage-cli clean"
+  },
+  "dependencies": {
+    "@backstage/catalog-model": "^0.6.0",
+    "@backstage/core": "^0.4.3",
+    "@backstage/theme": "^0.2.2",
+    "@backstage/plugin-catalog": "^0.2.7",
+    "@backstage/plugin-kafka-backend": "^0.1.0",
```

Comment:
This could be problematic to include the backend as a dependency for the frontend plugin. I also ran into this for the `rollbar` plugin but decided to duplicate code until a better solution was available. A common package could work, but this is also a lot extra work. Ideally there is an established & blessed pattern for sharing code between a backend/frontend plugin. Thoughts @backstage/maintainers?

Reply:
So this was also done in the [Kubernetes plugin](https://github.com/backstage/backstage/blob/master/plugins/kubernetes/package.json). I'm not sure which one I like more. Copying over sounds to me more error prone. Why do you think this is problematic?

Reply:
Backend plugins can contain node specific code that might not run on the frontend. It's possible this pattern is being used in some other packages and slipped through the cracks. I've had this conversation with @backstage/maintainers a couple of times and there has been some push back on doing this for good reason. However, like I said, there is no great alternative at this point unfortunately.

Reply:
Yep, agreed.

Remember that everything you import gets pulled into your package; it pollutes your package and punishes the users on the browser where package size and composition is an important factor. In node land we can wildly slap on packages without worrying too much, but in the other direction things are worse.

We sometimes break things out frontend+backend shared stuff into isomorphic packages (like the `packages/integrations` package) but that takes patience and keeping one's wits about oneself to get it right.